services/server/server.go

```diff
@@ -737,6 +741,44 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 	}, nil
 }
 
+// TestEmailAlertingSettings tests email alerting SMTP settings by sending testing email.
+func (s *Server) TestEmailAlertingSettings(
+	ctx context.Context,
+	req *serverpb.TestEmailAlertingSettingsRequest,
+) (*serverpb.TestEmailAlertingSettingsResponse, error) {
+	settings := &models.EmailAlertingSettings{
+		From:       req.Settings.From,
+		Smarthost:  req.Settings.Smarthost,
+		Hello:      req.Settings.Hello,
+		Username:   req.Settings.Username,
+		Password:   req.Settings.Password,
+		Identity:   req.Settings.Identity,
+		Secret:     req.Settings.Secret,
+		RequireTLS: req.Settings.RequireTls,
+	}
+
+	if err := settings.Validate(); err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Invalid argument: %s.", err.Error()))
+	}
+
+	if !govalidator.IsEmail(req.EmailTo) {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid \"emailTo\" email %q", req.EmailTo))
+	}
+
+	var errInvalidArgument *models.ErrInvalidArgument
+	err := s.emailer.Send(ctx, settings, req.EmailTo)
+	switch {
+	case err == nil:
+	case errors.As(err, &errInvalidArgument):
+		return nil, status.Error(codes.InvalidArgument,
+			fmt.Sprintf("Cannot send email: %s.", errInvalidArgument.Details))
+	default:
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Cannot send email: %s.", err.Error()))
+	}
```

Comment:
IMHO that looks much more simple:
```suggestion
	if err != nil {
		if errors.As(err, &errInvalidArgument) {
			return nil, status.Errorf(codes.InvalidArgument, "Cannot send email: %s.", errInvalidArgument.Details)
		}

		return nil, status.Errorf(codes.Internal, "Cannot send email: %s.", err.Error())
	}
```

Reply:
done